provisioner/puppet-masterless/provisioner.go

```diff
@@ -259,7 +259,13 @@ func (p *Provisioner) uploadManifests(ui packer.Ui, comm packer.Communicator) (s
 	}
 	defer f.Close()
 
-	manifestFilename := filepath.Base(p.config.ManifestFile)
+	manifestFilename := p.config.ManifestFile
+	if fi, err := os.Stat(p.config.ManifestFile); err != nil {
+		return "", fmt.Errorf("Error inspecting manifest file: %s", err)
+	} else if !fi.IsDir() {
```

Comment:
This looks like a bug, but it's actually intentional for BC reasons. Can we add a comment to note that?

Also docs should probably be updated to reflect the same.